genai.go

```diff
@@ -354,9 +383,109 @@ func (s *AgentServiceOp) CreateAgent(ctx context.Context, create *AgentCreateReq
 	return root.Agent, resp, nil
 }
 
+// ListAPIKeys retrieves list of API Keys associated with the specified GenAI agent
+func (s *GenAIServiceOp) ListAPIKeys(ctx context.Context, agentId string, opt *ListOptions) ([]*ApiKeyInfo, *Response, error) {
+	path := fmt.Sprintf("%s/%s/api_keys", genAIBasePath, agentId)
+	req, err := s.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(agentAPIKeysRoot)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	if l := root.Links; l != nil {
+		resp.Links = l
+	}
+	if m := root.Meta; m != nil {
+		resp.Meta = m
+	}
+
+	return root.ApiKeys, resp, nil
+}
+
+// CreateAPIKey creates a new API key for the specified GenAI agent
+func (s *GenAIServiceOp) CreateAPIKey(ctx context.Context, agentId string, create *AgentCreateAPIKeyRequest) (*ApiKeyInfo, *Response, error) {
```

Comment:
```suggestion
func (s *GenAIServiceOp) CreateAgentAPIKey(ctx context.Context, agentId string, create *AgentCreateAPIKeyRequest) (*ApiKeyInfo, *Response, error) {
```
Let's namespace this with "Agent" incase there's another type of APIKey that can be created